### PR TITLE
psp-4505 add claims for document generation.

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

**Review by asanchezr — APPROVED** [review]

**Review by FuriousLlama — CHANGES_REQUESTED** [review]
Will need to be the added to 38_01 build

**Comment by devinleighsmith** [comment]
> Will need to be the added to 38_01 build

Will do, I need to make some minor updates here based on requirements confirmation with the BAs.

File: source/backend/dal/Security/Permissions.cs [hunk 1]
```diff
@@ -123,16 +123,19 @@ public enum Permissions
         [Display(GroupName = "document", Name = "document-admin", Description = "Can perform admin functions on documents.")]
         DocumentAdmin = 39,
 
+        [Display(GroupName = "document", Name = "generate-documents", Description = "Can generate documents.")]
+        GenerateDocuments = 40,
```
Comment (FuriousLlama) [inline source/backend/dal/Security/Permissions.cs:127]: Also, we need one for an Admin to add template documents.

Comment (devinleighsmith) [reply source/backend/dal/Security/Permissions.cs:127]: I think for the story for that one I just said that simply checking for the admin role is good enough for adding template documents. I don't see a case where another user type would be given access to that page.

**Comment by devinleighsmith** [comment]
This one is on hold until 38.01 is merged.

File: source/backend/dal/Security/Permissions.cs (hunk 1 above)
Comment (FuriousLlama) [reply source/backend/dal/Security/Permissions.cs:127]: Do you mean what is stated in the story? It mentions that the user has the role of Admin. There is no mention of the claims. For synchronization with Mayan, there is document-admin permission that we could use if we want to group both admin responsibilities together.

Comment (devinleighsmith) [reply source/backend/dal/Security/Permissions.cs:127]: yep, My idea is that we wouldn't use a claim at all and just use role membership.

**Comment by devinleighsmith** [comment]
@FuriousLlama given the decision we came to on the document admin claim, can you remove the changes requested flag? (once I fix the conflict of course).

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**